index.js

```diff
@@ -153,22 +154,19 @@
   //. > type.parse (type (Identity (0)))
   //. {namespace: 'my-package', name: 'Identity', version: 1}
   //. ```
-  type.parse = function parse(s) {
-    var namespace = null;
-    var name = s;
-    var version = 0;
-    var groups = pattern.exec (s);
-    if (groups != null) {
-      namespace = groups[1];
-      name = groups[2];
-      if (groups[3] != null) version = Number (groups[3]);
-    }
-    return {namespace: namespace, name: name, version: version};
+  type.parse = s => {
+    // eslint-disable-next-line no-sparse-arrays
+    const [, namespace, name, version] = pattern.exec (s) || [, null, s, null];
```

Comment:
Are you happy with this approach, or do you find it overly clever?

Reply:
It looks good to me. You could insert `null` at the first position of the fallback array to please eslint, without affecting the program.